packages/millicast-sdk/src/utils/Diagnostics.js

```diff
@@ -39,9 +40,14 @@ const getDiagnostics = (statsCount = MAX_STATS_HISTORY_SIZE) => {
   diagnostics.version = version
   diagnostics.timestamp = Date.now()
   diagnostics.userAgent = window?.navigator?.userAgent || 'No user agent available'
+  diagnostics.stats = [...tempStats]
 
-  if (diagnostics.stats.length && Number.isInteger(statsCount) && statsCount >= 0 && statsCount < MAX_STATS_HISTORY_SIZE) {
-    diagnostics.stats = diagnostics.stats.slice(-statsCount)
+  if (tempStats.length && Number.isInteger(statsCount) && statsCount >= 0 && statsCount < MAX_STATS_HISTORY_SIZE) {
+    if (statsCount === 0) {
+      diagnostics.stats = []
+    } else {
+      diagnostics.stats = tempStats.slice(-statsCount)
```

Comment:
```suggestion
      diagnostics.stats = tempStats.slice(Math.max(-statsCount, -MAX_STATS_HISTORY_SIZE))
```

Reply:
This would be resolved with the first check of `statsCount`